node/syncblock.go

```diff
@@ -368,7 +406,44 @@ func (localNode *LocalNode) initSyncing() {
 // StartSyncing starts block syncing from current local ledger until it gets to
 // block height stopHeight with block hash stopHash from given neighbors
 func (localNode *LocalNode) StartSyncing(syncStopHash common.Uint256, syncStopHeight uint32, neighbors []*RemoteNode) (bool, error) {
-	var err error
+	cs := chain.DefaultLedger.Store.(*store.ChainStore)
+
+	fastSyncHeight, err := cs.GetSyncRootHeight()
+	if err != nil {
+		log.Info("local sync root height not found:", err)
+	}
+	if config.Parameters.SyncMode != "fast" && fastSyncHeight > cs.GetHeight() {
+		log.Fatalf("invalid sync mode %s at height %d", config.Parameters.SyncMode, cs.GetHeight())
```

Comment:
This will cause problem if user stop node during fast sync, turn off fast sync and restart node

Reply:
User can not stop fast sync and restart as full sync, because old stat didn't not exist which needed by full sync